app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Admin::UsersController < Admin::BaseController
+
+  def new
+    @user = User.new
+    @charity = current_user.charities.first.name
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      @user.charities << current_user.charities.first
+      @user.roles << Role.find_by(name: "charity_admin")
+      flash[:success] = "Welcome, #{@user.username}!"
+      redirect_to admin_dashboard_path
+    else
+      flash.now[:warning] = @user.errors.full_messages.join(", ")
+      render :new
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attribute('email', user_params[:email]) && @user.update_attribute('cell', user_params[:cell])
+      flash[:success] = "Your updates have been saved"
+      if @user.admin?
+        redirect_to admin_dashboard_path
+      else
+        redirect_to dashboard_path
+      end
+    else
+      flash.now[:warning] = @user.errors.full_messages.join(", ")
+      render :edit
+    end
+  end
+
+  def destroy
```

Comment:
I agree with Anna. I think this is really an update. The action itself may not even need to be routed through a controller. We can maybe either make a poro or an action on the model that handles the status update here.